### PR TITLE
GUI save config

### PR DESCRIPTION
**Pull request by HEdingfield**
Helps address #45.

### CODE REVIEW

**Comment by HEdingfield** [comment]
One thing I'm wondering: currently I have it set up so the first option in an enum is the default value, and will be automatically selected in the choice boxes. In light of this, should we rearrange any of the enums?

File: src/com/rcv/JsonParser.java [hunk 1]
```diff
@@ -22,6 +23,7 @@
   // file access: read
   // returns: instance of the object parsed from json or null if there was a problem
   static <T> T parseObjectFromFile(String jsonFilePath, Class<T> valueType) {
+    // TODO: consolidate two return statements in this method into one?
```
Comment (tarheel) [inline src/com/rcv/JsonParser.java]: oh, yeah, please fix that while you're here, if you don't mind

File: src/com/rcv/Tabulator.java [hunk 2]
```diff
@@ -847,20 +848,56 @@ void doAudit(List<CastVoteRecord> castVoteRecords) {
...
+    TRANSFER_WHOLE_SURPLUS("transferWholeSurplus"),
+    TRANSFER_RULE_UNKNOWN("transferRuleUnknown");
+
+    private final String label;
+
+    MultiSeatTransferRule(String label) {
+      this.label = label;
+    }
+
+    static MultiSeatTransferRule getByLabel(String labelLookup) {
```
Comment (tarheel) [inline src/com/rcv/Tabulator.java]: Darn. When I saw that you were using this above, I was hoping that it was a built-in thing that we had overlooked. But anyway, yeah, this is better encapsulation.

File: src/com/rcv/JsonParser.java (hunk 1 above)
Comment (HEdingfield) [reply src/com/rcv/JsonParser.java]: Yep, done in the next PR.

**Comment by HEdingfield** [comment]
> Looks good. As before, I didn't look closely at the UI stuff.

Still definitely appreciate you checking for general good Java coding practices!

> Is it possible to have these selectors default to nothing or a dummy "Select one" value? If you don't want to go that route, I don't really know what the right default values would be. I'd rather force the user to consciously select each setting.

Yep, I can do this. Only inconsistency is that the optional choices all have default values, so those will be auto-filled even though they aren't needed...

**Comment by tarheel** [comment]
> Only inconsistency is that the optional choices all have default values, so those will be auto-filled even though they aren't needed...

OK. Once we have the whole UI in place, it'll be easier to reason about whether some of those defaults should be removed.

**Pull request merged**